### PR TITLE
appservice: arg-name related bug fixes in hostname commands

### LINKED ISSUE #2946: [App Service] get-external-ip command doesn't support standard "name" parameter
### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: pip nightly

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 

azure-cli (2.0.3+1.dev20170419)

acr (2.0.1+1.dev20170419)
acs (2.0.3+1.dev20170419)
appservice (0.1.3+1.dev20170419)
batch (2.0.1+1.dev20170419)
cloud (2.0.1+1.dev20170419)
component (2.0.1+1.dev20170419)
configure (2.0.3+1.dev20170419)
container (0.1.3+1.dev20170419)
core (2.0.3+1.dev20170419)
documentdb (0.1.3+1.dev20170419)
feedback (2.0.1+1.dev20170419)
find (0.1.2rc1+1.dev20170312)
iot (0.1.3+1.dev20170419)
keyvault (2.0.1+1.dev20170419)
lab (0.0.2+1.dev20170419)
monitor (0.0.2+1.dev20170419)
network (2.0.3+1.dev20170419)
nspkg (2.0.0+1.dev20170419)
profile (2.0.3+1.dev20170419)
redis (0.2.0+1.dev20170419)
resource (2.0.3+1.dev20170419)
role (2.0.2+1.dev20170419)
sql (2.0.1+1.dev20170419)
storage (2.0.3+1.dev20170419)
vm (2.0.3+1.dev20170419)

Python (Darwin) 2.7.13 (default, Apr  4 2017, 08:47:57) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]

**OS Version:** What OS and version are you using?  
Answer here: macOS

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: bash

---

### Description

The `az appservice web config hostname get-external-ip` has a `--webapp-name` parameter, instead of `--name`, which is inconsistent, but more importantly, means that it doesn't naturally pick up a default `web` setting like all of the other `az appservice web` commands do.

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #2946, #2947, #2949

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by lostintangent** [comment]
LGTM!

File: src/command_modules/azure-cli-appservice/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 Release History
 ===============
+0.1.4 (unreleased)
+++++++++++++++++++++
+* hostname command bug fixes (#2946, #2947, #2949)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/HISTORY.rst]: I wouldn't expect anyone to go cross-reference closed GitHub issues to figure out what this means. You should summarize the changes from a user perspective--especially the breaking change.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -238,36 +238,40 @@ def _filter_for_container_settings(settings):
...
-                                   slot)
+    result = list(_generic_site_operation(resource_group_name, webapp_name,
+                                          'list_host_name_bindings', slot))
+    for r in result:
+        r.name = r.name.split('/')[-1]
+    return result
 
 
-def get_external_ip(resource_group_name, webapp_name):
+def get_external_ip(resource_group_name, webapp):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Why the change from webapp_name to webapp?

Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Thanks for catching this. I briefly renamed from `webapp_name` to `weapp`, but didn't backed out cleanly
